Source/CarthageKit/Git.swift

```diff
@@ -55,7 +55,7 @@ public func launchGitTask(
 		.ignoreTaskData()
 		.mapError(CarthageError.taskError)
 		.map { data in
-			return String(data: data, encoding: .utf8)!
+			return String(data: data, encoding: .utf8)!.stripping(suffix: "\0")
```

Comment:
Could you extract this to a function: `parseGitOutput(_ data: Data) -> String` and add a test that covers this?

Reply:
Maybe we should replace the `"\0"`s with `"\n"`s?

Reply:
@mdiep in [the case of `gitmodulesEntriesInRepository`](https://github.com/Carthage/Carthage/blob/00431497a74104880a391cd13427fe1949d731dd/Source/CarthageKit/Git.swift#L414) (and others) [we’re actually requesting null characters](https://github.com/Carthage/Carthage/blob/00431497a74104880a391cd13427fe1949d731dd/Source/CarthageKit/Git.swift#L414) — not as terminators, but as data separators (because paths are allowed to contain new lines).

Best (IMO) to leave the data unchanged at the `launchGitTask` level (once we Unicode decode it, of course) to not break what some caller expects to receive for parsing.